Sources/Testing/Traits/Trait.swift

```diff
@@ -41,6 +41,68 @@ public protocol Trait: Sendable {
   ///
   /// By default, the value of this property is an empty array.
   var comments: [Comment] { get }
+
+  /// The type of the custom test executor for this trait.
+  ///
+  /// The default type is `Never`.
+  associatedtype CustomTestExecutor: CustomTestExecuting = Never
+
+  /// The custom test executor for this trait, if any.
+  ///
+  /// If this trait's type conforms to ``CustomTestExecuting``, the default
+  /// value of this property is `self` and this trait will be used to customize
+  /// test execution. This is the most straightforward way to implement a trait
+  /// which customizes the execution of tests.
+  ///
+  /// However, if the value of this property is an instance of another type
+  /// conforming to ``CustomTestExecuting``, that instance will be used to
+  /// perform custom test execution instead.  Otherwise, the default value of
+  /// this property is `nil` (with the default type `Never?`), meaning that
+  /// custom test execution will not be performed for tests this trait is
+  /// applied to.
+  var customTestExecutor: CustomTestExecutor? { get }
+}
+
+/// A protocol that allows customizing the execution of a test function (and
```

Comment:
This comment becomes high-level documentation, so it would be good to add some explanation of what this means, what it's for… basically distill the pitch down into a couple of paragraphs.